osf/models/user.py

```diff
@@ -1528,6 +1534,32 @@ def get_node_comment_timestamps(self, target_id):
         default_timestamp = dt.datetime(1970, 1, 1, 12, 0, 0, tzinfo=pytz.utc)
         return self.comments_viewed_timestamp.get(target_id, default_timestamp)
 
+    def _get_spam_content(self, saved_fields):
```

Comment:
This function feels very structure-specific, as in it'll crash and burn if we use it on something other than what's send by the current endpoints for checking spam, aka employment and jobs that happen to be a list of nested dicts. 

Is this OK and we'll fix it as we go if we send more stuff? Can this be more efficient and/or improved?

Reply:
At a minimum, I think this function could be renamed to signify it only handles profile fields, but I'd like to think more on it.

Reply:
That's a good idea! `_get_spam_content` is specifically overriding a thing from `SpamMixin` which is why it's named that way, but there's no reason it couldn't just call a better named function.